addons/stock/models/stock_inventory.py

```diff
@@ -319,6 +324,32 @@ def _get_inventory_lines_values(self):
             vals += self._get_exhausted_inventory_lines_vals({(l['product_id'], l['location_id']) for l in vals})
         return vals
 
+    @api.model
+    def _run_inventory_tasks(self, company_id=False):
+        """ Generate and clean up inventories when certain conditions are met.
+        Creation conditions include:
+            - A location has a cyclic count set (i.e. inventory every XX days)
+        Deleting conditions include:
+            - Location's stock inventories not started before current day (i.e. still draft) [new one created]
+        """
+        domain = [('active', '=', True), ('next_inventory_date', '<=', fields.Date.today())]
+        if company_id:
+            # manually triggered => only apply to user's companies
+            domain = expression.AND([[('company_id', '=', company_id)], domain])
+        else:
+            # cron triggered => apply to all companies
+            domain = expression.AND([[('company_id', '!=', False)], domain])
+        locations = self.env['stock.location'].search(domain)
+        if locations:
+            self.search([('state', '=', 'draft'), ('location_ids', 'in', locations.ids), ('product_ids', '=', False), ('create_date', '<', fields.Datetime.today())]).unlink()
+            locations_w_inv = self.search([('state', 'in', ['draft', 'confirm']), ('location_ids', 'in', locations.ids), ('product_ids', '=', False)]).mapped('location_ids')
+            # Don't create overlapping inventories for a location + its children or new inventories when they've already been made
+            locations = locations.filtered(lambda l: l.location_id not in locations and l not in locations_w_inv)
+            for location in locations:
```

Comment:
Make sure you deal this raise here (you wont see usererror raised by the scheduler)
https://github.com/odoo/odoo/blob/341c644fed9dc67ef92d0bda6e4d2b0c650f0a6d/addons/stock/models/stock_inventory.py#L476-L490